plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/fs/lock/FileLockController.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.model.fs.lock;
+
+import com.google.gson.Gson;
+import org.jkiss.code.NotNull;
+import org.jkiss.code.Nullable;
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.Log;
+import org.jkiss.dbeaver.utils.GeneralUtils;
+import org.jkiss.utils.IOUtils;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.UUID;
+
+/**
+ * File based resource locks
+ */
+public class FileLockController {
+    private static final Log log = Log.getLog(FileLockController.class);
+    private static final long DEFAULT_MAX_LOCK_TIME = Duration.ofMinutes(1).toMillis(); // 1 min
+    private static final int CHECK_PERIOD = 10;
+
+    private static final String LOCK_META_FOLDER = ".locks";
+    private static final String LOCK_FILE_EXTENSION = ".lock";
+
+    private final Gson gson = new Gson();
+
+    private final Path lockFolderPath;
+    private final String applicationId;
+    private final long maxLockTime;
+
+    public FileLockController(@NotNull String applicationId) throws DBException {
+        this(applicationId, DEFAULT_MAX_LOCK_TIME);
+    }
+
+    // for tests
+    public FileLockController(@NotNull String applicationId, long maxLockTime) throws DBException {
```

Comment:
What the hell? So we're now polluting the storage during tests. Can we please not and instead make it properly testable? By the way, I don't see any tests.

Reply:
> So we're now polluting the storage during tests

nope, because for tests used test workspace which is automatically cleared

>  I don't see any tests.

tests still in cloudbeaver repo, to reduce the code in the pr